explorer/testdata/array/fail_to_deduce_nested_array_size_from_tuple_different_subtuples_size.carbon

```diff
@@ -0,0 +1,14 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+package ExplorerTest api;
+
+fn Main() -> i32 {
+  // CHECK:STDERR: COMPILATION ERROR: fail_to_deduce_nested_array_size_from_tuple_different_subtuples_size.carbon:[[@LINE+1]]: type pattern '{{\[\[}}i32;];]' does not match actual type '((i32, i32), (i32, i32, i32))'
```

Comment:
Plain
```
type pattern '[[i32;];]'
```
doesn't work, test fails with an error
```
Unexpected [[, should be {{\[\[}} at `i32;]`
```
I believe it's because of the FileCheck's substitution blocks syntax, so regex used to avoid the confusion.
If I'm missing something and there's a better way to verify the output, would appreciate any guidance.